hcl2template/types.packer_config.go

```diff
@@ -97,9 +106,15 @@ func (cfg *PackerConfig) EvalContext(variables map[string]cty.Value) *hcl.EvalCo
 				"cwd":  cty.StringVal(strings.ReplaceAll(cfg.Cwd, `\`, `/`)),
 				"root": cty.StringVal(strings.ReplaceAll(cfg.Basedir, `\`, `/`)),
 			}),
-			dataAccessor: cty.ObjectVal(datasourceVariables),
 		},
 	}
+
+	switch ctx {
```

Comment:
I think it'd be nice to add a comment here, too:
```suggestion
	
	// Currently the places where you can make references to other blocks
	// from one is very 'procedural', and in this specific case, we could make
	//  the data sources available to other datasources, but this would be
	// order dependant, meaning that if you define two datasources in two
	// different blocks, the second one can use the first one, but not the 
	// other way around; which would be totally confusing; so - for now -
	// datasources can't use other datasources.
	// In the future we'd like to load and execute HCL blocks using a graph
	// dependency tree, so that any block can use any block whatever the
	// order.
	switch ctx {
```

Reply:
Nice! Good idea, much more clear why we are doing this now. 👍🏼